### PR TITLE
Token flow sample

### PR DESCRIPTION
**Pull request by hassler-d**
Added a SAP-Java-Buildpack sample that is uses the token client lib without spring.

### CODE REVIEW

File: samples/sap-java-buildpack-tokenflow-usage/README.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+```xml
+<web-app>
+<display-name>sample</display-name>
+  <login-config> 
+    <auth-method>XSUAA</auth-method>
+  </login-config> 
+</web-app> 
+```
+
+In the Java coding, use the `@ServletSecurity` annotations:
```
Comment (nenaraab) [inline samples/sap-java-buildpack-tokenflow-usage/README.md]: delete as well...

File: samples/sap-java-buildpack-tokenflow-usage/README.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+		String clientid = extractString(jsonObject, "/xsuaa/0/credentials/clientid");
+		String url = extractString(jsonObject, "/xsuaa/0/credentials/url");
+
+		XsuaaTokenFlows tokenFlows = new XsuaaTokenFlows(
+				new DefaultOAuth2TokenService(),
+				new XsuaaDefaultEndpoints(url), new ClientCredentials(clientid, clientSecret));
+		OAuth2TokenResponse tokenResponse = tokenFlows.clientCredentialsTokenFlow().execute();
+
+		writeLine(response, "Access-Token: " + tokenResponse.getAccessToken());
+		writeLine(response, "Expired-At: " + tokenResponse.getExpiredAtDate());
```
Comment (nenaraab) [inline samples/sap-java-buildpack-tokenflow-usage/README.md]: additionally you can return / print tokenResponse.getDecodedAccessToken.getPayload() ....

File: samples/sap-java-buildpack-tokenflow-usage/README.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+Finally, as part of your Identity Provider, e.g. SAP ID Service, assign the deployed Role Collection(s) such as `Tokenflow_Buildpack_API_Viewer` to your user as depicted in the screenshot below and as documented [here](https://help.sap.com/viewer/65de2977205c403bbc107264b8eccf4b/Cloud/en-US/9e1bf57130ef466e8017eab298b40e5e.html).
+
+![](../images/SAP_CP_Cockpit_AssignRoleCollectionToUser.png)
+
+Further up-to-date information you can get on sap.help.com:
+- [Maintain Role Collections](https://help.sap.com/viewer/65de2977205c403bbc107264b8eccf4b/Cloud/en-US/d5f1612d8230448bb6c02a7d9c8ac0d1.html)
+- [Maintain Roles for Applications](https://help.sap.com/viewer/65de2977205c403bbc107264b8eccf4b/Cloud/en-US/7596a0bdab4649ac8a6f6721dc72db19.html).
+
+## Access the application
+After deployment, the application router will trigger authentication. If you have assigned the role-collection provided in the [xs-security.json](./xs-security.json) to your user, you will see an output like when calling `https://approuter-sap-java-buildpack-tokenflow-usage-<<ID>>.<<LANDSCAPE_APPS_DOMAIN>>`:
```
Comment (nenaraab) [inline samples/sap-java-buildpack-tokenflow-usage/README.md]: take care that you do not access the service via approuter url https://approuter-sap-java-buildpack-tokenflow-usage ...

File: samples/sap-java-buildpack-tokenflow-usage/README.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+```
+Access-Token: eyJhbGciOiJSUzI1NiIsImprdSI6I...
+Expired-At: Tue Oct 15 02:55:27 UTC 2019
+```
+If not you should get a `403` status code (Forbidden).
+
+> Note: you can find the route of your approuter application using `cf app approuter-sap-java-buildpack-tokenflow-usage`.
```
Comment (nenaraab) [inline samples/sap-java-buildpack-tokenflow-usage/README.md]: delete

File: samples/sap-java-buildpack-tokenflow-usage/manifest.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  routes:
+    - route: sap-java-buildpack-tokenflow-usage-((ID)).((LANDSCAPE_APPS_DOMAIN))
+
+  host: 
+  path: target/sap-java-buildpack-tokenflow-usage.war
+  buildpacks:
+    - sap_java_buildpack
+  services:
+    - xsuaa-tokenflow
+# Application Router as web server
```
Comment (nenaraab) [inline samples/sap-java-buildpack-tokenflow-usage/manifest.yml]: delete ...

**Comment by hassler-d** [comment]
I don't understand why the build failed. I cannot re-trigger it manually, can you do that @nenaraab?

**Pull request merged**